src/util.debounce.js

```diff
@@ -0,0 +1,20 @@
+function debounce({callback, cancelIf, delay}) {
+  let timeoutId = null;
+
+  return (...args) => {
+    timeoutId && clearTimeout(timeoutId);
+
+    if (cancelIf(...args)) {
+      if (timeoutId) {
+        timeoutId = null;
+      }
+      return
+    }
+
+    timeoutId = setTimeout(() => {
+      clearTimeout(timeoutId);
+      callback(...args);
+      timeoutId = null;
+    }, delay);
+  }
```

Comment:
Missing semicolon. Recommend installing prettifier-atom if you're using Atom. Then choose "Format files on save".

Reply:
@nphi1212 whenever you've finished addressing a comment, please click "Resolve conversation" (if it also appears to you like it does to me).